tests/components/template/conftest.py

```diff
@@ -105,6 +120,163 @@ async def async_setup_modern_trigger_format(
     await hass.async_block_till_done()
 
 
+class TemplateEntityTestsSetup:
```

Comment:
I am not sure how big of a fan I am of having such helper class, I don't think we have many of them and would love a second opinion on them. Currently tests are generally overseeable without a lot of state, and having such object could change that. But I might not have a full picture of all pros and cons here